doc/computing/randomness.rst

```diff
@@ -0,0 +1,345 @@
+`RandomState` instance will influence each other, as we will see later when
+we discuss cloning. This point is important to keep in mind when debugging.
+
+If we had passed an integer to the `random_state` parameter of the
+:class:`~sklearn.ensemble.RandomForestClassifier`, we would have obtained the
+same models, and thus the same scores each time. When we pass an integer, the
+same RNG is used across all calls to `fit`. What internally happens is that
+even though the RNG is consumed when `fit` is called, it is always reset to
+its original state at the beginning of `fit`.
+
+CV splitters
+............
+
+Randomized CV splitters have a similar behavior when a `RandomState`
+instance is passed; calling `split` multiple times yields different data
+splits::
+
+    >>> from sklearn.model_selection import KFold
+    >>> import numpy as np
+
+    >>> X = y = np.arange(10)
+    >>> rng = np.random.RandomState(0)
+    >>> cv = KFold(n_splits=2, shuffle=True, random_state=rng)
+
+    >>> for train, test in cv.split(X, y):
+    ...     print(train, test)
+    [0 3 5 6 7] [1 2 4 8 9]
+    [1 2 4 8 9] [0 3 5 6 7]
+
+    >>> for train, test in cv.split(X, y):
+    ...     print(train, test)
+    [0 4 6 7 8] [1 2 3 5 9]
+    [1 2 3 5 9] [0 4 6 7 8]
+
+We can see that the splits are different from the second time `split` is
+called. This may lead to unexpected results if you compare the performance of
+multiple estimators by calling `split` many times, as we will see in the next
+section.
+
+Common pitfalls and subtleties
+------------------------------
+
+While the rules that govern the `random_state` parameter are seemingly simple,
+they do however have some subtle implications. In some cases, this can even
+lead to wrong conclusions.
+
+Estimators
+..........
+
+**Different `random_state` types lead to different cross-validation
+procedures**
+
+Depending on the type of the `random_state` parameter, estimators will behave
+differently, especially in cross-validation procedures. Consider the
+following snippet::
+
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.model_selection import cross_val_score
+    >>> import numpy as np
+
+    >>> X, y = make_classification(random_state=0)
+
+    >>> rf_123 = RandomForestClassifier(random_state=123)
+    >>> cross_val_score(rf_123, X, y)
+    array([0.85, 0.95, 0.95, 0.9 , 0.9 ])
+
+    >>> rf_inst = RandomForestClassifier(random_state=np.random.RandomState(0))
+    >>> cross_val_score(rf_inst, X, y)
+    array([0.9 , 0.95, 0.95, 0.9 , 0.9 ])
+
+We see that the cross-validated scores of `rf_123` and `rf_inst` are
+different, as should be expected since we didn't pass the same `random_state`
+parameter. However, the difference between these scores is more subtle than
+it looks, and **the cross-validation procedures that were performed by**
+:func:`~sklearn.model_selection.cross_val_score` **significantly differ in
```

Comment:
```suggestion
it looks, and **the cross-validation splits that were performed by**
:func:`~sklearn.model_selection.cross_val_score` **differ in
```

Reply:
I'm not sure I agree with the suggested change, because the splits are the same in this code snippet. What differs is the cross-validation procedure, as described just below.

Reply:
Yes, you are right. Brain fart, my mistake.